MonoGame.Framework/Audio/SoundEffect.cs

```diff
@@ -17,32 +17,63 @@ public sealed partial class SoundEffect : IDisposable
     {
         #region Internal Audio Data
 
-        private string _name;
+        private string _name = string.Empty;
         
         private bool _isDisposed = false;
-        private TimeSpan _duration = TimeSpan.Zero;
+        private readonly TimeSpan _duration = TimeSpan.Zero;
 
         #endregion
 
         #region Internal Constructors
 
-        internal SoundEffect() { }
+        internal SoundEffect() {}
+
+        internal SoundEffect(byte[] buffer, int format, int sampleRate, int channels, int blockAlignment, int durationMs, int loopStart, int loopLength)
+        {
+            _duration = TimeSpan.FromMilliseconds(durationMs);
+
+            // This is regular PCM data.
+            if (format == 1)
+            {
+                PlatformInitializePCM(buffer, 0, buffer.Length, sampleRate, (AudioChannels) channels, loopStart, loopLength);
+                return;
+            }
+
+            // Everything else is platform specific.
+            PlatformInitializeFormat(buffer, format, sampleRate, channels, blockAlignment, loopStart, loopLength);
+        }
 
         #endregion
 
         #region Public Constructors
 
-        /// <param name="buffer">Buffer containing PCM wave data.</param>
+        /// <param name="buffer">Buffer containing 16bit PCM wave data.</param>

```

Comment:
It would be good to extend this for the platforms that can support higher fidelity, such as 24-bit PCM and floating point. It has been brought up in comments before.